src/lapack/backends/cusolver/cusolver_batch.cpp

```diff
@@ -79,251 +100,645 @@ void getri_batch(sycl::queue &queue, std::int64_t n, sycl::buffer<std::complex<d
+
+            // Transform ptr and stride to list of ptr's
+            cuDataType **a_batched = create_ptr_list_from_stride(a_, stride_a, batch_size);
+            CUDA_ERROR_FUNC(cuMemAlloc, cuda_result, &a_dev, sizeof(T *) * batch_size);
+            CUDA_ERROR_FUNC(cuMemcpyHtoD, cuda_result, a_dev, a_batched, sizeof(T *) * batch_size);
+
+            auto **a_dev_ = reinterpret_cast<cuDataType **>(a_dev);
+
+            CUSOLVER_ERROR_FUNC_T_SYNC(func_name, func, err, handle, get_cublas_fill_mode(uplo),
+                                       (int)n, a_dev_, (int)lda, nullptr, (int)batch_size);
+
+            free(a_batched);
+            cuMemFree(a_dev);
+        });
+    });
 }
-void ungqr_batch(sycl::queue &queue, std::int64_t m, std::int64_t n, std::int64_t k,
-                 sycl::buffer<std::complex<float>> &a, std::int64_t lda, std::int64_t stride_a,
-                 sycl::buffer<std::complex<float>> &tau, std::int64_t stride_tau,
-                 std::int64_t batch_size, sycl::buffer<std::complex<float>> &scratchpad,
-                 std::int64_t scratchpad_size) {
-    throw unimplemented("lapack", "ungqr_batch");
+
+// Scratchpad memory not needed as parts of buffer a is used as workspace memory
+#define POTRF_STRIDED_BATCH_LAUNCHER(TYPE, CUSOLVER_ROUTINE)                                      \
+    void potrf_batch(sycl::queue &queue, oneapi::mkl::uplo uplo, std::int64_t n,                  \
+                     sycl::buffer<TYPE> &a, std::int64_t lda, std::int64_t stride_a,              \
+                     std::int64_t batch_size, sycl::buffer<TYPE> &scratchpad,                     \
+                     std::int64_t scratchpad_size) {                                              \
+        return potrf_batch(#CUSOLVER_ROUTINE, CUSOLVER_ROUTINE, queue, uplo, n, a, lda, stride_a, \
+                           batch_size, scratchpad, scratchpad_size);                              \
+    }
+
+POTRF_STRIDED_BATCH_LAUNCHER(float, cusolverDnSpotrfBatched)
+POTRF_STRIDED_BATCH_LAUNCHER(double, cusolverDnDpotrfBatched)
+POTRF_STRIDED_BATCH_LAUNCHER(std::complex<float>, cusolverDnCpotrfBatched)
+POTRF_STRIDED_BATCH_LAUNCHER(std::complex<double>, cusolverDnZpotrfBatched)
+
+#undef POTRF_STRIDED_BATCH_LAUNCHER
+
+template <typename Func, typename T>
+inline void potrs_batch(const char *func_name, Func func, sycl::queue &queue,
+                        oneapi::mkl::uplo uplo, std::int64_t n, std::int64_t nrhs,
+                        sycl::buffer<T> &a, std::int64_t lda, std::int64_t stride_a,
+                        sycl::buffer<T> &b, std::int64_t ldb, std::int64_t stride_b,
+                        std::int64_t batch_size, sycl::buffer<T> &scratchpad,
+                        std::int64_t scratchpad_size) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+
+    overflow_check(n, nrhs, lda, ldb, stride_a, stride_b, batch_size, scratchpad_size);
+
+    // cusolver function only supports nrhs = 1
```

Comment:
cuSolver is referred with different letter casing across the comments.

Reply:
Done